waffle-provider/src/MockProvider.ts

```diff
@@ -17,4 +41,12 @@ export class MockProvider extends providers.Web3Provider {
   createEmptyWallet() {
     return Wallet.createRandom().connect(this);
   }
+
+  clearCallHistory() {
+    this.callHistory = [];
+  }
+
+  getCallHistory(): readonly RecordedCall[] {
+    return this.callHistory;
+  }
 }
```

Comment:
Not sure if this should be put here, but we need some api to filter history by address and decode by ABI

Reply:
I would say that this should go in the matchers